presto-main/src/main/java/com/facebook/presto/execution/buffer/LazyOutputBuffer.java

```diff
@@ -312,6 +312,12 @@ public void fail()
         outputBuffer.fail();
     }
 
+    @Override
+    public synchronized long getPeakMemoryUsage()
+    {
+        return delegate.getPeakMemoryUsage();
```

Comment:
`delegate` can be null.  I'd copy the pattern of the code in the other methods like `isOverutilized`